ert/data/record/_record.py

```diff
@@ -165,17 +274,42 @@ def _infer_type(self, data: numerical_record_data) -> RecordType:
 
     @property
     def index(self) -> RecordIndex:
+        """Returns the data index.
+
+        Returns:
+            RecordIndex: Returns handle to data index.
+        """
         return self._index
 
     @property
     def data(self) -> record_data:
+        """Returns record data.
```

Comment:
Is this useful documentation? Ie:

```py
@property 
def foo(self) -> Foo:
    """Returns foo"""
```

Reply:
Maybe the documentation can go away for the Abstract class (es) then?